tests/unit/test_click_annotations.py

```diff
@@ -44,7 +44,30 @@ def _command_id_fn(val):
     return val.callback.__module__
 
 
+_ALL_PARSING_ATTRS = {
+    attrname: getattr(globus_cli.parsing, attrname)
+    for attrname in globus_cli.parsing.__all__
+}
+_ALL_CUSTOM_PARAM_TYPES = (
+    getattr(globus_cli.parsing, attrname)
+    for attrname, attrval in _ALL_PARSING_ATTRS.items()
+    if isinstance(attrval, type) and issubclass(attrval, click.ParamType)
+)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="test requires py3.10+")
+@pytest.mark.parametrize(
+    "param_type", _ALL_CUSTOM_PARAM_TYPES, ids=lambda x: x.__name__
+)
+def test_custom_param_types_are_annotated(param_type):
+    from click_type_test import AnnotatedParamType
+
+    assert isinstance(param_type, AnnotatedParamType)
```

Comment:
I think that testing for a specific Python version is slightly incorrect here.

What matters is that `click_type_test` is importable, not what version of Python the tests are running under. They're linked conditions, but I think it will be better to cut to the heart of the matter: "Is `click_type_test` installed?"

```suggestion
@pytest.mark.parametrize(
    "param_type", _ALL_CUSTOM_PARAM_TYPES, ids=lambda x: x.__name__
)
def test_custom_param_types_are_annotated(param_type):
    click_type_test = pytest.importorskip("click_type_test")

    assert isinstance(param_type, click_type_test.AnnotatedParamType)
```

This feedback applies to the next test function, too.

Reply:
I didn't know about importorskip! I'll look into exactly how I'd like to phrase this -- I'd sort of like a module level skip if possible -- but I agree with you about the boundary between checking that click-type-test is available and checking the python version.

Reply:
I _think_ that you can use `importorskip` as a module-level check, but I'll watch for the next update!

Reply:
It works quite well at the module-level, though not (IMO) perfectly.
We get a single "skip" reported in the text test report, but there's no specific skipped test output with `-v`. So if someone were trying to find which tests got skipped in a given run, it could get a little confusing.

Still, I like that we get to skip the _whole_ module this way, so if we wanted a future in which `click-type-test` helped us build test parameters, that would be very possible.